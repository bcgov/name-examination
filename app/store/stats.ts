import { getStats } from '~/util/namex-api'
import { getFormattedDateWithTime } from '~/util/date'
import { sortNameChoices } from '~/util'
import { type NameRequest } from '~/types'
import { defineStore } from 'pinia'

export const useStatsStore = defineStore('stats', () => {
  /**
   * Fetches paged statistics data based on the provided timespan and myStats flag.
   *
   * @param {string | null} timespan - The timespan for the statistics data. If null, the default timespan is used.
   * @param {boolean} myStats - A flag indicating whether to fetch statistics related to user's own stats.
   * @return {Promise<any>} A promise that resolves with the fetched statistics data.
   * @throws {Error} Throws an error if the fetching process fails.
   */
  async function getPagedStats (timespan: string | null, myStats: boolean): Promise<any> {
    const params = new URLSearchParams({
      timespan: timespan || '1',
      currentpage: '1',
      perpage: '1000'
    })
    if (myStats) params.append('myStats', 'True')

    try {
      const response = await getStats(params)

      if (response.ok) {
        const jsonData = await response.json() // Parse response JSON

        if (jsonData.numRecords >= 0) {
          const requestsData = jsonData.nameRequests as Array<NameRequest>

          const processedData = requestsData.map((statsData:any) => {
            statsData.names = sortNameChoices(statsData.names)
            if (statsData.comments.length === 1 && statsData.comments[0].timestamp) {
              statsData.comments[0].timestamp = getFormattedDateWithTime(statsData.comments[0].timestamp)
            }
            statsData.lastUpdate = getFormattedDateWithTime(statsData.lastUpdate)
            statsData.names.forEach((name:any) => {
              if (name.decision_text) {
                name.decision_text = name.decision_text.split(/\n\n/)
              }
            })
            return statsData
          })
          return {
            nameRequests: processedData,
            numRecords: jsonData.numRecords
          }
        }
      } else {
        throw new Error('Network response was not ok')
      }
    } catch (error) {
      console.error('There was a problem with your fetch operation:', error)
      throw error // Rethrow the error to handle it elsewhere if needed
    }
  }

  return {
    getPagedStats
  }
})
