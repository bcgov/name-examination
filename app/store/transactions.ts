import {
  ConsentFlag,
  type EntityTypeCode,
  type RequestActionCode,
  type RequestTypeCode,
} from '~/enums/codes'
import { Status } from '~/enums/nr-status'
import {
  type NameChoice,
  type NameRequest,
  type TransactionEntry,
  type Transactions,
} from '~/types'
import { emitter } from '~/util/emitter'
import { getNameRequest, getTransactions } from '~/util/namex-api'
import { toMappedRequestType } from '~/util/request-type'
import requestTypes from '~/data/request_types.json'
import { sortNameChoices } from '~/util'

export const useTransactions = defineStore('transactions', () => {
  const nr = ref<NameRequest>()
  const transactions = ref<Array<TransactionEntry>>([])
  const loadingNr = ref(false)
  const loadingTransactions = ref(false)

  async function loadNr(nrNumber: string) {
    loadingNr.value = true
    const nrResponse = await getNameRequest(nrNumber)
    nr.value = await nrResponse.json()
    loadingNr.value = false
  }

  async function loadTransactions(nrNumber: string) {
    loadingTransactions.value = true
    const transactionsResponse = await getTransactions(nrNumber)
    const transactionsJson = (await transactionsResponse.json()) as Transactions
    transactionsJson.transactions.forEach(t => sortNameChoices(t.names))
    transactions.value = transactionsJson.transactions
    loadingTransactions.value = false
  }

  async function initialize(nrNumber: string) {
    try {
      await loadNr(nrNumber)
      await loadTransactions(nrNumber)
    } catch (e) {
      emitter.emit('error', {
        title: 'Transactions Error',
        message: 'Failed to retrieve transactions data.',
      })
    } finally {
      loadingNr.value = false
      loadingTransactions.value = false
    }
  }

  function getStatusDisplay(status: Status | null, names: Array<NameChoice>) {
    if (!status) return 'N/A'

    let displayState = status as string
    if (status === Status.Conditional) {
      displayState = 'CONDITIONAL APPROVED'
    } else if (status === Status.Consumed && names && names.length > 0) {
      const approvedName = names.find((name) =>
        [Status.Approved, Status.Condition].includes(name.state)
      )
      displayState =
        approvedName?.state === Status.Condition
          ? 'CONDITIONAL APPROVED'
          : 'APPROVED'
      displayState += ` / Used for ${approvedName?.corpNum}`
    }
    return displayState
  }

  function getRequestTypeDisplay(
    requestType: RequestTypeCode,
    action: RequestActionCode
  ) {
    if (!nr.value) return 'N/A'
    const mapped = toMappedRequestType(
      requestType,
      action,
      nr.value.entity_type_cd
    )
    return requestTypes.find((r) => r.value == mapped)?.text || 'N/A'
  }

  function getConsentDisplay(date: string | null, consentFlag: ConsentFlag) {
    return date
      ? 'Required. Received.'
      : consentFlag === ConsentFlag.Required
      ? 'Required. Not yet received.'
      : 'Not required'
  }

  return {
    nr,
    transactions,
    loadingNr,
    loadingTransactions,
    initialize,
    getStatusDisplay,
    getRequestTypeDisplay,
    getConsentDisplay,
  }
})
