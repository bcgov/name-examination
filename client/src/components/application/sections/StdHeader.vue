<!--eslint-disable-->
<template>
  <div class="container-fluid">
    <nav class="navbar fixed-top navbar-dark navbar-expand-lg" id="header-main">

      <router-link to="/home" class="navbar-brand router-link-never-mark-active">
        <img src="static/images/gov3_bc_logo.png"
             alt="Province of British Columbia"
             title="Province of British Columbia logo"/>
        Namex
      </router-link>
      <button class="navbar-toggler"
              type="button"
              data-toggle="collapse"
              data-target="#navbarSupportedContent"
              aria-controls="navbarSupportedContent"
              aria-expanded="false"
              aria-label="Toggle navigation">
        <span class="navbar-toggler-icon"></span>
      </button>

      <!-- logged in -->
      <div v-if="auth" class="collapse navbar-collapse" id="navbarSupportedContent">

        <ul class="navbar-nav ml-auto">
          <li class="nav-item">
            <span id="userid" class="nav-link">{{ userId }}</span>
          </li>
          <li class="navbar-text divider">|</li>
          <li class="nav-item">
            <a id="header-logout-button" href="#" @click="onLogout" class="nav-link">Logout</a>
          </li>
        </ul>
      </div>

      <!-- not logged in -->
      <div v-else class="collapse navbar-collapse" id="navbarSupportedContent">
        <router-link tag="button" id="header-login-button" to="/signin" class="btn btn-sm ml-auto">Login</router-link>
      </div>

      <!-- SECONDARY MENU -->
      <div v-if="auth" class="navbar fixed-top" id="secondary-menu">

        <ul class="navbar-nav mr-auto">
          <li class="nav-item">
            <a target="_blank" href="https://namex-solr-dev.pathfinder.gov.bc.ca" id="admin" class="nav-link">Admin</a>
          </li>
          <li v-if="userCanExamine" class="navbar-text divider">|</li>
          <li v-if="userCanExamine" class="nav-item">
            <router-link to="/nameExamination" id="nameExamine" class="nav-link">Examine Names</router-link>
          </li>
          <li class="navbar-text divider">|</li>
          <li class="nav-item">
            <router-link to="/find" class="nav-link" id="header-search-link">Search</router-link>
          </li>
          <li class="navbar-text divider">|</li>
          <li class="nav-item">
            <form class="form-inline" id="header-search-form" @submit.prevent="onSubmit">
              <input id="header-search-input" class="form-control mr-sm-2" type="search"
                     placeholder="NR Number" aria-label="Search" v-model="nrNum">
              <button id="header-search-button" class="btn btn-outline-success my-2 my-sm-0"
                      type="submit">Load</button>
            </form>
          </li>
        </ul>

        <router-link to="/stats" class="nav-link ml-auto">Stats</router-link>

      </div>
    </nav>
  </div>
</template>

<script>
/* eslint-disable */
  export default {
    name: "std-header",
    data () {
      return {
        nrNum: ''
      }
    },
    computed: {
      userId() {
        return this.$store.getters.userId
      },
      auth() {
        return this.$store.getters.isAuthenticated
      },
      userCanExamine() {
        return this.$store.getters.userHasApproverRole
      }
    },
    methods: {
      onLogout() {
        this.$store.dispatch('logout')
        window.location.assign("/");
      },
      onSubmit() {
        if(this.nrNum!='') {
          var myNum = this.nrNum.toUpperCase().trim();
          if (myNum.includes('NR')) {
            if (!myNum.includes('NR ')) {
              myNum = myNum.replace('NR', 'NR ')
            }
          } else {
            myNum = 'NR ' + myNum
          }
          console.log('Set new NR number to:' + myNum)
          this.$store.dispatch('newNrNumber', myNum)
          this.nrNum = ''
          this.$router.push('/nameExamination')
        }
      }
    }
  }
</script>

<style scoped>

  .navbar {
    padding-left: 40px;
  }

  .navbar-brand img {
    margin-top: -13px; /* bump the image up, since it's got white/blue space in the image itself that
                          adds artificial padding */
    margin-right: 10px;
    margin-left: 10px;
    margin-bottom: 5px;
  }

  .nav-link {
    color: white !important;
  }


  #header-login-button {
    background-color: white;
    color: #003366;
  }

  #secondary-menu {
    height: 50px;
    background-color: #38598a;
    color: white;
    margin-top: 69px;
    box-shadow: 0px 5px 10px 0px rgba(169,169,169,1);
    font-size: 15px;
    line-height: 23px;
  }

  #secondary-menu .divider {
    color: #7597d6;
  }

  #header-search-button {
    border-color: #003366;
    color: white;
    padding-top: 3px;
    padding-bottom: 3px;
  }
  #header-search-button:hover {
    border-color: #003366;
    background-color: #1a5a96;
  }

  #header-search-form {
    padding-right: .5rem;
    padding-left: .5rem;
  }

</style>
