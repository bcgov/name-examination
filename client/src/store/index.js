/* eslint-disable */
import Vue from 'vue'
import Vuex from 'vuex'
import axios from '../axios-auth'
import globalAxios from 'axios'
import router from '@/router'

Vue.use(Vuex)

export default new Vuex.Store({
  state: {
    nrNum: null,
    authorized: null,
    userId: null,
    email: null,
    kctoken: null,
    nr_conflict: null,
    user_role: null,
    currentChoice: null,
    currentName: null,
    currentState: null,
    is_editing: false,
    nrData: null,
    details: null,
   // formData: {
      compInfo: {
        nrNumber: null,
        compNames: {
          compName1: null,
          compName2: null,
          compName3: null
        },
        requestType: null,
      },
      applicantInfo: {
        applicantname: {
          firstName: null,
          lastname: null
        },
        contactInfo: {
          address: null,
          contactName: null,
          phone: null,
          email: null,
          fax: null
        },
      },
      additionalCompInfo: {
        jurisdiction: null,
        natureOfBussiness: null,
        nuans: null,
        sk_name: null,
        nr_status: null
      },
      examiner: null,
      priority: null,
      reSubmission: {
        reSubmissionYN: null,
        linkedNR: null
      },
      reservationCount: null,
      expiryDate: null,
  //  },
    issueText: null,
    issue: {
        issue_Match: null,
        issue_Consent: null,
        issue_TradeMark: null,
        issue_History: null,
        issue_Format: null,
        issue_Match_Text: null,
        issue_Consent_Text: null,
        issue_TradeMark_Text: null,
        issue_History_Text: null,
        issue_Format_Text: null
      }
  },

  mutations: {
    authUser (state, userData) {
      state.kctoken = userData
      //state.kctoken = userData.client_session
      //state.userRole = userData.user_role
      //state.idToken = userData.client_session
    },

    clearAuthData (state) {
      state.kcToken = null
      state.userId = null
    },

    setIssueText (state,issue) {
      switch (issue) {
        case "M1":
          state.issueText = state.issue.issue_Match_Text;
          break;
        case "M2":
          state.issueText = state.issue.issue_Consent_Text;
          break;
        case "M3":
          state.issueText = state.issue.issue_TradeMark_Text;
          break;
        case "M4":
          state.issueText = state.issue.issue_History_Text;
          break;
        case "M5":
          state.issueText = state.issue.issue_Format_Text;
          break;
        default:
          state.issueText = state.issue.issue_Match_Text;
      }
    },

    loadpostgresNo(state, postgresData) {
        state.compInfo.nrNumber = postgresData.nameRequest
    },

    loadCompanyInfo(state, dbcompanyInfo) {
      console.log('Loading Company Info into state')

      state.nrData = dbcompanyInfo

      if (dbcompanyInfo.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of dbcompanyInfo.names) {
          switch (record.choice) {
            case 1:
              state.compInfo.compNames.compName1 = record.name
              break;
            case 2:
              state.compInfo.compNames.compName2 = record.name
              break;
            case 3:
              state.compInfo.compNames.compName3 = record.name
              break;
            default:
              console.log('Error with company name structure')
              return
          }
        }
      }

      state.compInfo.requestType = dbcompanyInfo.requestTypeCd
      state.applicantInfo.applicantname.firstName = dbcompanyInfo.applicant
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.applicantInfo.contactInfo.contactName = dbcompanyInfo.contact
      state.applicantInfo.contactInfo.phone = dbcompanyInfo.phoneNumber
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.additionalCompInfo.jurisdiction = dbcompanyInfo.xproJurisdiction
      state.additionalCompInfo.natureOfBussiness = dbcompanyInfo.natureBusinessInfo
      state.additionalCompInfo.nuans = dbcompanyInfo.nuansNum
      state.additionalCompInfo.sk_name = dbcompanyInfo.skPartner
      state.additionalCompInfo.nr_status = dbcompanyInfo.state
      state.examiner = dbcompanyInfo.userId
      state.priority = dbcompanyInfo.priorityCD
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.expiryDate = dbcompanyInfo.expiryDate
    },

    loadCompanyIssues(state, dbcompanyIssues) {
      //state.issue.issue_Match = dbcompanyIssues.isue_Match
      //state.issue.issue_Match_Text = dbcompanyIssues.issue_Match_Text
      //state.issue.issue_Consent = dbcompanyIssues.issue_Consent
      //state.issue.issue_Consent_Text = dbcompanyIssues.issue_Consent_Text
      //state.issue.issue_TradeMark = dbcompanyIssues.issue_TradeMark
      //state.issue.issue_TradeMark_Text = dbcompanyIssues.issue_TradeMark_Text
      //state.issue.issue_History = dbcompanyIssues.issue_History
      //state.issue.issue_History_Text = dbcompanyIssues.issue_History_Text
      //state.issue.issue_Format = dbcompanyIssues.issue_Format
      //state.issue.issue_Format_Text = dbcompanyIssues.issue_Format_Text
    },

    update_nrData(state){

      if(state.nrData.names.length == 0) {
        console.log('Error, No company names found')
        return
      } else {
        for (let record of state.nrData.names) {
          switch (record.choice) {
            case 1:
              state.nrData.names[0].name = state.compInfo.compNames.compName1
              break;
            case 2:
              state.nrData.names[0].name = state.compInfo.compNames.compName2
              break;
            case 3:
              state.nrData.names[0].name = state.compInfo.compNames.compName3
              break;
          }
        }
      }

      state.nrData.requestTypeCd = state.compInfo.requestType
      state.nrData.applicant = state.applicantInfo.applicantname.firstName
      //state.applicantInfo.applicantname.lastname = dbcompanyInfo.lastName
      //state.applicantInfo.contactInfo.address = dbcompanyInfo.address
      state.nrData.contact = state.applicantInfo.contactInfo.contactName
      state.nrData.phoneNumber =  state.applicantInfo.contactInfo.phone
      //state.applicantInfo.contactInfo.email = dbcompanyInfo.email
      //state.applicantInfo.contactInfo.fax = dbcompanyInfo.fax
      state.nrData.xproJurisdiction = state.additionalCompInfo.jurisdiction
      state.nrData.natureBusinessInfo = state.additionalCompInfo.natureOfBussiness
      state.nrData.nuansNum = state.additionalCompInfo.nuans
      state.nrData.skPartner = state.additionalCompInfo.sk_name
      //state.nrData.state =  state.additionalCompInfo.nr_status
      state.nrData.userId = state.examiner
      state.nrData.priorityCD = state.priority
      //state.reSubmission.reSubmissionYN = dbcompanyInfo.resubmissionYN
      //state.reSubmission.linkedNR = dbcompanyInfo.linkedNR
      //state.reservationCount = dbcompanyInfo.reservationCount
      state.nrData.expiryDate = state.expiryDate
    },

    saveDetail(state,detail){
      state.details = detail
    }
  },

  actions: {
    kcauth ({commit, dispatch, state}) {
      const kc = localStorage.getItem('KEYCLOAK_TOKEN')
      state.kctoken = kc
      //console.log(authData)
      //state.user_role =  authData.realm_access.roles,
      state.user_role = ''
      state.idToken = kc
      localStorage.setItem('kctoken',state.kctoken)
      localStorage.setItem('user_role',state.user_role)

      if (state.kctoken){
        console.log('KC Authorized user roles')
        commit('authUser', {
          //user_role: state.user_role,
          kctoken: state.kctoken
        })
      }
    },

    logout ({commit,state}) {

      commit('clearAuthData')

      localStorage.removeItem('expirationDate')
      localStorage.removeItem('token')
      localStorage.removeItem('user_role')
      localStorage.removeItem('userId')
      localStorage.removeItem('email')
      localStorage.removeItem('COMPINFO')

      localStorage.removeItem('kctoken')
      localStorage.removeItem('KEYCLOAK_TOKEN')
      localStorage.removeItem('AUTHORIZED')
      localStorage.removeItem('USERNAME')

    },

    tryAutoLogin ({commit}) {
    },

    storeUser ({commit, state}, userData) {
      if (!state.idToken) {
        return
      }
      globalAxios.post('/users.json' + '?auth=' + state.idToken, userData)
        .then(res => console.log(res))
        .catch(error => console.log(error))
    },

    fetchUser ({commit, state}) {
      if (!state.idToken) {
        return
      }
      globalAxios.get('/users.json' + '?auth=' + state.idToken)
        .then(res => {
          console.log(res)
          const data = res.data
          const users = []
          for (let key in data) {
            const user = data[key]
            user.id = key
            users.push(user)
          }
          console.log(users)
          commit('storeUser', users[0])
        })
        .catch(error => console.log(error))
    },

    setDetails({commit, state}) {
     var detail = state.details?null:"1";
     commit('saveDetail',detail)
    },

    selectNameIssue ({commit, dispatch, state}, divID) {
      console.log('action: selected issue control:' + divID)
      localStorage.setItem('issueText',divID )
      commit('setIssueText', divID)
    },

    getpostgrescompNo ({commit, dispatch, state}) {
      console.log('action: select next company number from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/queues/@me/oldest'
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp No Response:' + response)
        commit('loadpostgresNo',response.data)
      })
    },

    getpostgrescompInfo ({commit},nrNumber) {
      console.log('action: getting data for company number: ' + nrNumber + ' from postgres')
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + nrNumber
      console.log('URL:' + url)
      const vm = this
      return axios.get(url, {headers: {Authorization: `Bearer ${myToken}`}}).then(response => {
        console.log('Comp Info Response:' + response.data)
        localStorage.setItem('COMPINFO',response.data)
        commit('loadCompanyInfo',response.data)
      })
      .catch(error => console.log('ERROR: ' + error))
    },

    updateNRState ({state},nrState) {
      console.log('Updating Examination state for number: ' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber

      axios.patch(url,{headers: {Authorization: `Bearer ${myToken}`}},{"state": nrState} )
           .then(function(response){
                console.log('state updated to ' + nrState + ' for ' + state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
      },

    nameApproved( {commit, state}) {
      console.log('Name Approved for ' + state.compInfo.nrNumber + ", " + state.currentName)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'APPROVED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url, state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name approved for ' + state.compInfo.nrNumber)
           })
           .catch(error => console.log('ERROR: ' + error))
    },

    nameRejected({commit, state}) {
      console.log('Name Rejected for NR #' + state.compInfo.nrNumber)
      const myToken = localStorage.getItem('KEYCLOAK_TOKEN')
      state.nrData.state = 'REJECTED'
      commit('update_nrData')
      const url = '/api/v1/requests/' + state.compInfo.nrNumber
      axios.put(url,state.nrData, {headers: {Authorization: `Bearer ${myToken}`}})
           .then(function(response){
              console.log('Name rejected for ' +  state.compInfo.nrNumber)
            })
            .catch(error => console.log('ERROR: ' + error))
    }
  },

  getters: {
    is_editing(state) {
      return state.is_editing
    },
    user(state) {
      return state.user
    },
    email(state) {
      return state.email
    },
    issueText(state) {
      return state.issueText
    },
    isAuthenticated(state) {
      return localStorage.getItem("AUTHORIZED")
    },
    nrNumber(state) {
      return state.compInfo.nrNumber
    },
    compName1(state) {
     return state.compInfo.compNames.compName1
    },
    compName2(state) {
      return state.compInfo.compNames.compName2
    },
    compName3(state) {
      return state.compInfo.compNames.compName3
    },
    requestType(state) {
      return state.compInfo.requestType
    },
    firstName(state) {
      return state.applicantInfo.applicantname.firstName
    },
    lastName(state) {
      return state.applicantInfo.applicantname.lastname
    },
    address(state) {
      return state.applicantInfo.contactInfo.address
    },
    conEmail(state) {
      return  state.applicantInfo.contactInfo.email
    },
    contactName(state) {
      return state.applicantInfo.contactInfo.contactName
    },
    phone(state) {
      return state.applicantInfo.contactInfo.phone
    },
    fax(state) {
      return state.applicantInfo.contactInfo.fax
    },
    jurisdiction(state) {
      return  state.additionalCompInfo.jurisdiction
    },
    natureOfBusiness(state) {
      return state.additionalCompInfo.natureOfBussiness
    },
    nuans(state) {
      return state.additionalCompInfo.nuans
    },
    sk_name(state) {
      return state.additionalCompInfo.sk_name
    },
    nr_status(state) {
      return state.additionalCompInfo.nr_status
    },
    examiner(state) {
      return state.examiner
    },
    priority(state) {
      return state.priority
    },
    reSubmissionYN(state) {
      return state.reSubmission.reSubmissionYN
    },
    linkedNR(state) {
      return state.reSubmission.linkedNR
    },
    reservationCount(state) {
      return state.reservationCount
    },
    expiryDate(state) {
      return state.expiryDate
    },
    issue_Match(state) {
      return state.issue.issue_Match
    },
    issue_Match_Text(state) {
      return state.issue.issue_Match_Text
    },
    issue_Consent(state) {
      return state.issue.issue_Consent
    },
    issue_Consent_Text(state) {
      return state.issue.issue_Consent_Text
    },
    issue_TradeMark(state) {
      return state.issue.issue_TradeMark
    },
    issue_TradeMark_Text(state) {
      return state.issue.issue_TradeMark_Text
    },
    issue_History(state) {
      return state.issue.issue_History
    },
    issue_History_Text(state) {
      return state.issue.issue_History_Text
    },
    issue_Format(state) {
      return state.issue.issue_Format
    },
    issue_Format_Text(state) {
      return state.issue.issue_Format_Text
    },
    details(state) {
      return state.details
    },
    addInfo(state) {
      return state.details
    }
  }
})
