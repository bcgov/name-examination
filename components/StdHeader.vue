<template>
  <nav class="flex bg-white shadow-2xl font-medium h-20 text-lg">
      <div class="flex min-w-full">
        
          <div class="image">
              <img src="../public/images/top-nav.png" alt="Name Examination" class="h-full">
          </div>

          <div class="flex gap-24 mt-6 text-blue-800" v-if="this.auth">
              <a href="" class="ml-16 text-2xl" >Admin</a>
              <a href="" class="text-2xl">Examine</a>
              <a href="" class="text-2xl"><nuxt-link to="/search">Search</nuxt-link></a>
          </div>

          <div class="flex ml-auto" v-if="this.auth">

            <form class="flex items-center">   
              <label for="allowWordClassificationModal" class="sr-only">NR Number Lookup</label>
              <div class="relative w-full">
                  <div class="absolute inset-y-0 left-0 flex items-center pl-3 pointer-events-none">
                      <svg aria-hidden="true" class="w-7 h-7 text-gray-500 dark:text-gray-400" fill="currentColor" viewBox="0 0 20 20" xmlns="http://www.w3.org/2000/svg"><path fill-rule="evenodd" d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z" clip-rule="evenodd"></path></svg>
                  </div>
                  <input type="text" id="allowWordClassificationModal" class="w-72 h-14 bg-gray-50 border border-gray-300 text-gray-900 rounded-lg focus:ring-blue-500 focus:border-blue-500 block pl-10 p-2.5  dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500 text-lg" placeholder="NR Number Lookup" required>
              </div>
              <button type="submit" class="p-2.5 text-sm font-medium text-white bg-blue-700 rounded-lg border border-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800">
                  <img src="../public/images/searchbutton.png" alt="" class="h-8 w-9">
              </button>
             </form>

             <span class="mt-7 mx-12 underline">Stats</span>
             
            <label class="relative inline-flex items-center cursor-pointer mt-2">
              <input type="checkbox" value="" class="sr-only peer">
              <div class="w-11 h-6 bg-gray-200 peer-focus:outline-none peer-focus:ring-4 peer-focus:ring-blue-300 dark:peer-focus:ring-blue-800 rounded-full peer dark:bg-gray-700 peer-checked:after:translate-x-full peer-checked:after:border-white after:content-[''] after:absolute after:left-[2px] after:top-[25px] after:bg-white after:border-gray-300 after:border after:rounded-full after:h-5 after:w-5 after:transition-all dark:border-gray-600 peer-checked:bg-blue-600"></div>
              <span class="ml-3 font-medium text-gray-500 dark:text-gray-300">Classify Words</span>
            </label>

            <span class="flex mt-7 mx-10 text-gray-800">mkapoor@idir</span>

            <div class="divider"></div>

            <div class="mx-10 mt-7 text-blue-800">
              <a href="">Log Out</a>
             </div>
          </div>

          <div class="flex ml-auto mx-10 mt-7 text-blue-800" v-if="!this.auth">
              <a href="">Login</a>
             </div>

      </div>
  </nav>
</template>

<script>
export default{
  data(){
    return{
      auth: true
    }
  },
  computed:{

  }
}
</script>