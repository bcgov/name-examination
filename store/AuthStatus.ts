import { defineStore } from 'pinia';

export const useAuthStore = defineStore('auth', {
  state: () => ({
    auth: false,
  }),
  actions: {
    toggleAuth() {
      this.auth = !this.auth;
    },
  },
});
