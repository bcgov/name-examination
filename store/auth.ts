import ConfigHelper from '../util/config-helper'
import KeycloakServices from '../public/keycloak/keycloak'
import { SessionStorageKeys } from '../util/constants'
import { defineStore } from 'pinia'

interface AuthState {
  token: string;
  idToken: string;
  refreshToken: string;
  kcGuid: string;
  loginSource: string;
}

export const useAuthStore = defineStore({
  id: 'auth',
  state: (): AuthState => ({
    token: '',
    idToken: '',
    refreshToken: '',
    kcGuid: '',
    loginSource: ''
  }),
  getters: {
    isAuthenticated (state): boolean {
      return !!state.token
    },
    keycloakGuid (state): string {
      return state.kcGuid || KeycloakServices.getUserInfo().keycloakGuid
    },
    currentLoginSource (state): string {
      return state.loginSource || KeycloakServices.getUserInfo().loginSource
    }
  },
  actions: {
    setKCToken (token: string): void {
      this.token = token
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakToken, token)
    },
    setIDToken (idToken: string): void {
      this.idToken = idToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakIdToken, idToken)
    },
    setRefreshToken (refreshToken: string): void {
      this.refreshToken = refreshToken
      ConfigHelper.addToSession(SessionStorageKeys.KeyCloakRefreshToken, refreshToken)
    },
    setKCGuid (kcGuid: string): void {
      this.kcGuid = kcGuid
    },
    setLoginSource (loginSource: string): void {
      this.loginSource = loginSource
    },
    clearSession (): void {
      // Reset all state properties related to the user's session
      this.token = ''
      this.idToken = ''
      this.refreshToken = ''
      this.kcGuid = ''
      this.loginSource = ''

      // Clear the session storage values
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakIdToken)
      ConfigHelper.removeFromSession(SessionStorageKeys.KeyCloakRefreshToken)
    },
    syncWithSessionStorage (): void {
      const token = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakToken) || ''
      const idToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakIdToken) || ''
      const refreshToken = ConfigHelper.getFromSession(SessionStorageKeys.KeyCloakRefreshToken) || ''

      this.token = token
      this.idToken = idToken
      this.refreshToken = refreshToken
    }
  }
})

if (import.meta.hot) {
  import.meta.hot.accept(acceptHMRUpdate(useAuthStore, import.meta.hot))
}
