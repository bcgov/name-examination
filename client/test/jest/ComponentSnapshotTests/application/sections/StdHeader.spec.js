/* eslint-disable */
import { shallowMount } from '@vue/test-utils';
import StdHeader from '@/components/application/sections/StdHeader';
import store from '@/store'

describe('StdHeader.vue', () => {

    let component;
    beforeEach(() => {
        component = shallowMount(StdHeader, {store: store});
    });

   it("renders a StdHeader component", () => {
     expect(component.element).toMatchSnapshot();
   })

  // Add other tests specific to this component and not its sub-components
});
