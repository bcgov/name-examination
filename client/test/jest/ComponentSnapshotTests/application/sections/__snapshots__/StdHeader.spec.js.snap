// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StdHeader.vue renders a StdHeader component 1`] = `
<div
  class="container-fluid"
>
  <nav
    class="navbar fixed-top navbar-dark navbar-expand-lg"
    id="header-main"
  >
    <router-link-stub
      class="navbar-brand router-link-never-mark-active"
      event="click"
      tag="a"
      to="/home"
    >
      <img
        alt="Province of British Columbia"
        src="static/images/gov3_bc_logo.png"
        title="Province of British Columbia logo"
      />
      
      Namex
    
    </router-link-stub>
     
    <button
      aria-controls="navbarSupportedContent"
      aria-expanded="false"
      aria-label="Toggle navigation"
      class="navbar-toggler"
      data-target="#navbarSupportedContent"
      data-toggle="collapse"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
     
    <div
      class="collapse navbar-collapse"
      id="navbarSupportedContent"
    >
      <router-link-stub
        class="btn btn-sm ml-auto"
        event="click"
        id="header-login-button"
        tag="button"
        to="/signin"
      >
        Login
      </router-link-stub>
    </div>
     
    <!---->
  </nav>
</div>
`;
